# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsRange.cpp
# Commit: 260574c2a78e
# Full Hash: 260574c2a78e619e6c9815a797a8df7086004994
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Update ranges to correctly set and update nodes for flattened selection r=jjaschke,smaug,dom-core"
#   
#   This reverts commit 0221ba097eb5884f3efe76c607cb0cc7c9170635.
# ==============================================================================

diff -r d699f1b53a1e -r 260574c2a78e dom/base/nsRange.cpp
--- a/dom/base/nsRange.cpp	Thu May 08 22:53:05 2025 +0300
+++ b/dom/base/nsRange.cpp	Thu May 08 22:53:06 2025 +0300
@@ -87,31 +87,26 @@
 
 template already_AddRefed<nsRange> nsRange::Create(
     const RangeBoundary& aStartBoundary, const RangeBoundary& aEndBoundary,
-    ErrorResult& aRv, AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+    ErrorResult& aRv);
 template already_AddRefed<nsRange> nsRange::Create(
     const RangeBoundary& aStartBoundary, const RawRangeBoundary& aEndBoundary,
-    ErrorResult& aRv, AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+    ErrorResult& aRv);
 template already_AddRefed<nsRange> nsRange::Create(
     const RawRangeBoundary& aStartBoundary, const RangeBoundary& aEndBoundary,
-    ErrorResult& aRv, AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+    ErrorResult& aRv);
 template already_AddRefed<nsRange> nsRange::Create(
     const RawRangeBoundary& aStartBoundary,
-    const RawRangeBoundary& aEndBoundary, ErrorResult& aRv,
-    AllowRangeCrossShadowBoundary aAlloCrossShadowBoundary);
-
+    const RawRangeBoundary& aEndBoundary, ErrorResult& aRv);
+
+template nsresult nsRange::SetStartAndEnd(const RangeBoundary& aStartBoundary,
+                                          const RangeBoundary& aEndBoundary);
+template nsresult nsRange::SetStartAndEnd(const RangeBoundary& aStartBoundary,
+                                          const RawRangeBoundary& aEndBoundary);
 template nsresult nsRange::SetStartAndEnd(
-    const RangeBoundary& aStartBoundary, const RangeBoundary& aEndBoundary,
-    AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
-template nsresult nsRange::SetStartAndEnd(
-    const RangeBoundary& aStartBoundary, const RawRangeBoundary& aEndBoundary,
-    AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
-template nsresult nsRange::SetStartAndEnd(
-    const RawRangeBoundary& aStartBoundary, const RangeBoundary& aEndBoundary,
-    AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+    const RawRangeBoundary& aStartBoundary, const RangeBoundary& aEndBoundary);
 template nsresult nsRange::SetStartAndEnd(
     const RawRangeBoundary& aStartBoundary,
-    const RawRangeBoundary& aEndBoundary,
-    AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+    const RawRangeBoundary& aEndBoundary);
 
 template void nsRange::DoSetRange(const RangeBoundary& aStartBoundary,
                                   const RangeBoundary& aEndBoundary,
@@ -214,13 +209,11 @@
 template <typename SPT, typename SRT, typename EPT, typename ERT>
 already_AddRefed<nsRange> nsRange::Create(
     const RangeBoundaryBase<SPT, SRT>& aStartBoundary,
-    const RangeBoundaryBase<EPT, ERT>& aEndBoundary, ErrorResult& aRv,
-    AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary) {
+    const RangeBoundaryBase<EPT, ERT>& aEndBoundary, ErrorResult& aRv) {
   // If we fail to initialize the range a lot, nsRange should have a static
   // initializer since the allocation cost is not cheap in hot path.
   RefPtr<nsRange> range = nsRange::Create(aStartBoundary.GetContainer());
-  aRv = range->SetStartAndEnd(aStartBoundary, aEndBoundary,
-                              aAllowCrossShadowBoundary);
+  aRv = range->SetStartAndEnd(aStartBoundary, aEndBoundary);
   if (NS_WARN_IF(aRv.Failed())) {
     return nullptr;
   }